### PR TITLE
feat: differentiate config map secret in deployment stage

### PR DESCRIPTION
**Pull request by rish2320**
# Description

Create a differentiating label to differentiate config maps and secrets in both edit deployment and new deployment workflow ( in both pre and post deployment stage ) , on selecting a config name or a secret having same name as the other type both are disappearing.

Fixes # [2532](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=2532)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?
1: To start with the , created label to differentiate config maps and secrets in the deployment stage.
2: On selecting the config map with secret , the latter does not disappear and vice versa.
3: The value of both config maps and secrets are passed correctly in the payload.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/cdPipeline/CDPipeline.tsx [hunk 1]
```diff
@@ -879,6 +881,19 @@ export default class CDPipeline extends Component<CDPipelineProps, CDPipelineSta
             key === 'preStage'
                 ? this.state.pipelineConfig.runPreStageInEnv
                 : this.state.pipelineConfig.runPostStageInEnv
+        function getOptionLabel(option:any) {
```
Comment (milindk8) [inline src/components/cdPipeline/CDPipeline.tsx:884]: Please assign proper type if possible

File: src/components/cdPipeline/CDPipeline.tsx [hunk 2]
```diff
@@ -879,6 +881,19 @@ export default class CDPipeline extends Component<CDPipelineProps, CDPipelineSta
...
+            if (option.type === 'configmaps') {
+                return (<div style={{ display: 'flex', alignItems: 'center', marginLeft: '0px'}}>
```
Comment (milindk8) [inline src/components/cdPipeline/CDPipeline.tsx:886]: please move all the inline CSS in the function to scss file & try to use classes

File: src/components/cdPipeline/CDPipeline.tsx [hunk 3]
```diff
@@ -912,8 +927,8 @@ export default class CDPipeline extends Component<CDPipelineProps, CDPipelineSta
                         isClearable={true}
                         value={selections}
                         options={this.configMapAndSecrets}
-                        getOptionLabel={(option) => `${option.name}`}
-                        getOptionValue={(option) => `${option.name}`}
+                        formatOptionLabel={getOptionLabel}
+                        getOptionValue={(option) => `${option.name}${option.type}`}
```
Comment (milindk8) [inline src/components/cdPipeline/CDPipeline.tsx:931]: create a function & then pass it here

File: src/components/cdPipeline/CDPipeline.tsx [hunk 4]
```diff
@@ -912,8 +927,8 @@ export default class CDPipeline extends Component<CDPipelineProps, CDPipelineSta
...
                         onChange={(selected) => {
                             this.handleConfigmapAndSecretsChange(selected, configmapKey)
```
Comment (milindk8) [inline src/components/cdPipeline/CDPipeline.tsx:933]: please do here as well

File: src/components/cdPipeline/CDPipeline.tsx (hunk 2 above)
Comment (rish2320) [reply src/components/cdPipeline/CDPipeline.tsx:886]: done

File: src/components/cdPipeline/CDPipeline.tsx (hunk 3 above)
Comment (rish2320) [reply src/components/cdPipeline/CDPipeline.tsx:931]: done

File: src/components/cdPipeline/CDPipeline.tsx (hunk 4 above)
Comment (rish2320) [reply src/components/cdPipeline/CDPipeline.tsx:933]: done

File: src/components/cdPipeline/CDPipeline.tsx (hunk 1 above)
Comment (rish2320) [reply src/components/cdPipeline/CDPipeline.tsx:884]: The type was not defined.

File: src/components/cdPipeline/CDPipeline.tsx [hunk 5]
```diff
@@ -879,6 +881,27 @@ export default class CDPipeline extends Component<CDPipelineProps, CDPipelineSta
             key === 'preStage'
                 ? this.state.pipelineConfig.runPreStageInEnv
                 : this.state.pipelineConfig.runPostStageInEnv
+        function getOptionLabel(option) {
```
Comment (milindk8) [inline src/components/cdPipeline/CDPipeline.tsx:884]: format file

**Review by milindk8 — APPROVED** [review]

File: src/components/cdPipeline/CDPipeline.tsx (hunk 5 above)
Comment (rish2320) [reply src/components/cdPipeline/CDPipeline.tsx:884]: done

File: src/components/cdPipeline/CDPipeline.tsx [hunk 6]
```diff
@@ -879,6 +881,27 @@ export default class CDPipeline extends Component<CDPipelineProps, CDPipelineSta
...
+            if (option.type === 'configmaps') {
```
Comment (vivek-devtron) [inline src/components/cdPipeline/CDPipeline.tsx:885]: We should use const of configmaps

File: src/components/cdPipeline/cdpipeline.util.tsx [hunk 7]
```diff
@@ -22,22 +22,31 @@ export const styles = {
             color: 'var(--N900)'
         })
     },
+    multiValue: (base ,state) => {
```
Comment (vivek-devtron) [inline src/components/cdPipeline/cdpipeline.util.tsx:25]: The same style has been used on 2 places, please verify on both places

File: src/components/cdPipeline/CDPipeline.tsx [hunk 8]
```diff
@@ -879,6 +881,27 @@ export default class CDPipeline extends Component<CDPipelineProps, CDPipelineSta
...
+                return (<div className='dropdown__option'>
+                    <File className='icon-dim-16' />
+                    <span style={{ marginLeft: 8, alignItems: 'center', fontSize: '12px' }}>{option.name}</span>
```
[the remark this answers is not in the document]
Comment (rish2320) [reply src/components/cdPipeline/CDPipeline.tsx]: done

File: src/components/cdPipeline/CDPipeline.tsx (hunk 6 above)
Comment (rish2320) [reply src/components/cdPipeline/CDPipeline.tsx]: done

File: src/components/cdPipeline/cdpipeline.util.tsx (hunk 7 above)
Comment (rish2320) [reply src/components/cdPipeline/cdpipeline.util.tsx:25]: done

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**